### ISSUE TITLE
UNRAID app

### ISSUE THREAD

**Issue by ragaimeena** [LibrePhotos/librephotos#276]
**Describe the enhancement you'd like**

Any plans on making this available on Unraid CA repository?

**Describe why this will benefit the LibrePhotos**
A clear and concise explanation on why this will make LibrePhotos better.
This will make it more popular and adopted easily.

**Additional context**
Add any other context or screenshots about the enhancement request here.
Thank You

**Comment by jeremytodd1** [comment]
Agree! Having this on Unraid's Community Apps page would be very helpful.

**Comment by derneuere** [comment]
I cannot do this as I do not have an unraid system. Follow this guide [here](https://forums.unraid.net/topic/57181-docker-faq/#comment-566084), send me the xml and then I can write the admin.

**Comment by Cobbert** [comment]
I have an unraid system, I will read the following and try to provide the needed xml.

**Comment by Cobbert** [comment]
I think I did it correctly, but please let me know if changes are needed, its an easy process to get the XML.

XML
```
***Removed XML to prevent confusion***
```

Unraid CMD
`***removed to prevent confusion*** `

**Comment by derneuere** [comment]
Thank you very much for this contribution! 

I think it should be based on this image here: https://hub.docker.com/r/reallibrephotos/singleton
The singleton image wraps all the different containers into one container, which is more in line with other Unraid Apps.

Here is the documentation on what you can supply to it: https://docs.librephotos.com/1/standard_install/#docker

**Comment by Cobbert** [comment]
Perfect, I will compile a new XML using the data provided and post it back here shortly.

**Comment by Cobbert** [comment]
Here is updated XML
```

<?xml version="1.0"?>
<Container version="2">
  <Name>Librephotos</Name>
  <Repository>reallibrephotos/singleton</Repository>
  <Registry>https://hub.docker.com/r/reallibrephotos/singleton</Registry>
  <Network>bridge</Network>
  <MyIP/>
  <Shell>sh</Shell>
  <Privileged>false</Privileged>
  <Support>https://github.com/LibrePhotos/librephotos/issues</Support>
  <Project>https://github.com/LibrePhotos/librephotos</Project>
  <Overview>Features&#xD;
&#xD;
    Support for all types of photos including raw photos&#xD;
    Support for videos&#xD;
    Timeline view&#xD;
    Scans pictures on the file system&#xD;
    Multiuser support&#xD;
    Generate albums based on events like "Thursday in Berlin"&#xD;
    Face recognition / Face classification&#xD;
    Reverse geocoding&#xD;
    Object / Scene detection&#xD;
    Semantic image search&#xD;
    Search by metadata&#xD;
&#xD;
What does it use?&#xD;
&#xD;
    Image Conversion: ImageMagick&#xD;
    Video Conversion: FFmpeg&#xD;
    Exif Support: ExifTool&#xD;
    Face detection: face_recognition&#xD;
    Face classification/clusterization: scikit-learn and hdbscan&#xD;
    Image captioning: im2txt,&#xD;
    Scene classification places365&#xD;
    Reverse geocoding: Mapbox: You need to have an API key. First 50,000 geocode lookups are free every month.&#xD;
</Overview>
  <Category>Cloud: MediaApp:Photos MediaServer:Photos Status:Beta</Category>
  <WebUI>localhost:3000</WebUI>
  <TemplateURL/>
  <Icon>https://github.com/LibrePhotos/librephotos/blob/89beca99d57f4987a68af61099ecc46396ffdcda/screenshots/site-logo.png</Icon>
  <ExtraParams/>
  <PostArgs/>
  <CPUset/>
  <DateInstalled>1659901858</DateInstalled>
  <DonateText/>
  <DonateLink/>
  <Description>Features&#xD;
&#xD;
    Support for all types of photos including raw photos&#xD;
    Support for videos&#xD;
    Timeline view&#xD;
    Scans pictures on the file system&#xD;
    Multiuser support&#xD;
    Generate albums based on events like "Thursday in Berlin"&#xD;
    Face recognition / Face classification&#xD;
    Reverse geocoding&#xD;
    Object / Scene detection&#xD;
    Semantic image search&#xD;
    Search by metadata&#xD;
&#xD;
What does it use?&#xD;
&#xD;
    Image Conversion: ImageMagick&#xD;
    Video Conversion: FFmpeg&#xD;
    Exif Support: ExifTool&#xD;
    Face detection: face_recognition&#xD;
    Face classification/clusterization: scikit-learn and hdbscan&#xD;
    Image captioning: im2txt,&#xD;
    Scene classification places365&#xD;
    Reverse geocoding: Mapbox: You need to have an API key. First 50,000 geocode lookups are free every month.&#xD;
</Description>
  <Networking>
    <Mode>bridge</Mode>
    <Publish>
      <Port>
        <HostPort>3000</HostPort>
        <ContainerPort>80</ContainerPort>
        <Protocol>tcp</Protocol>
      </Port>
    </Publish>
  </Networking>
  <Data>
    <Volume>
      <HostDir>/mnt/user/appdata/librephotos/photos</HostDir>
      <ContainerDir>/var/lib/librephotos/photos/</ContainerDir>
      <Mode>rw</Mode>
    </Volume>
    <Volume>
      <HostDir>/mnt/user/appdata/librephotos/thumbnails</HostDir>
      <ContainerDir>/var/lib/librephotos/data/protected_media/</ContainerDir>
      <Mode>rw</Mode>
    </Volume>
    <Volume>
      <HostDir>/mnt/user/appdata/librephotos/logs</HostDir>
      <ContainerDir>/var/log/librephotos/</ContainerDir>
      <Mode>rw</Mode>
    </Volume>
    <Volume>
      <HostDir>/mnt/user/appdata/librephotos/bd</HostDir>
      <ContainerDir>/var/lib/postgresql/data/</ContainerDir>
      <Mode>rw</Mode>
    </Volume>
  </Data>
  <Environment/>
  <Labels/>
  <Config Name="Photos" Target="/var/lib/librephotos/photos/" Default="" Mode="rw" Description="Container Path: /var/lib/librephotos/photos/" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/librephotos/photos</Config>
  <Config Name="thumbnails" Target="/var/lib/librephotos/data/protected_media/" Default="" Mode="rw" Description="Container Path: /var/lib/librephotos/data/protected_media/" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/librephotos/thumbnails</Config>
  <Config Name="logs" Target="/var/log/librephotos/" Default="" Mode="rw" Description="Container Path: /var/log/librephotos/" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/librephotos/logs</Config>
  <Config Name="database" Target="/var/lib/postgresql/data/" Default="" Mode="rw" Description="Container Path: /var/lib/postgresql/data/" Type="Path" Display="always" Required="true" Mask="false">/mnt/user/appdata/librephotos/bd</Config>
  <Config Name="Host Port 1" Target="80" Default="" Mode="tcp" Description="Container Port: 80" Type="Port" Display="always" Required="true" Mask="false">3000</Config>
</Container>
```

Unraid CMD

`/usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker create --name='Librephotos' --net='bridge' -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" -p '3000:80/tcp' -v '/mnt/user/appdata/librephotos/photos':'/var/lib/librephotos/photos/':'rw' -v '/mnt/user/appdata/librephotos/thumbnails':'/var/lib/librephotos/data/protected_media/':'rw' -v '/mnt/user/appdata/librephotos/logs':'/var/log/librephotos/':'rw' -v '/mnt/user/appdata/librephotos/bd':'/var/lib/postgresql/data/':'rw' 'reallibrephotos/singleton' `

**Comment by Cobbert** [comment]
Report back if anything else needs updating from unraid itself, or if you get this up and I will test it out on my box.

**Comment by derneuere** [comment]
Alright, I wrote the admin and opened up a thread: https://forums.unraid.net/topic/126943-support-librephotos/
Lets see if it goes online :)

**Comment by Cobbert** [comment]
`Unlike commercial service that store your photos in the cloud and scan/index them to train their machine learning models and collect ad targeting data on you, LibrePhotos keeps all your photos and metadata on your local machine. Your data is never sent to or stored on a 3rd party server. Get the same power as those commercial services without giving up your personal data and privacy.`

Just want to say, this is beautiful and why I am into open sourced, self hosted systems. Very well put, I am sure the unraid world will respond well.

**Comment by Cobbert** [comment]
Installing the docker to my unraid server now, will report back if it goes well.

**Comment by Cobbert** [comment]
Pulling the template, it looks like it used the original XML I posted since it pulled reallibrephotos/librephotos rather then reallibrephotos/singleton, this didn't have any of the containers directories configured by default nor the port mapping I added in the second XML posting.

CMD that was ran,
`root@localhost:# /usr/local/emhttp/plugins/dynamix.docker.manager/scripts/docker run -d --name='LibrePhotos' --net='bridge' -e TZ="America/Los_Angeles" -e HOST_OS="Unraid" 'reallibrephotos/librephotos:latest'`

This CMD comes from the original posting rather then the second posting.

**Comment by derneuere** [comment]
Alright, do I have to add another file for the CMD command to the repo?

You can also find the file here: https://github.com/LibrePhotos/librephotos-unraid

**Comment by derneuere** [comment]
Oh I copied the wrong one 🤦

**Comment by Cobbert** [comment]
Haha I figured it was just the wrong XML, I am not sure about the CMD, that may have nothing to do with the repo and may be something generated when you add the docker to unraid, but when I generated the XML, it provided it and I figured I should add it along with the XML in case you needed it.

I have never added anything to unraid CA app so im not sure on the proccess.

**Comment by Cobbert** [comment]
Updated my other post to keep the thread a bit more clean

**Comment by jeremytodd1** [comment]
Do these updates mean a listing for LibrePhotos is pretty close to happening on Unraid's Community Apps?

**Comment by Cobbert** [comment]
It was up, but a mixup with the xml happened, so I think it is being updated now, I have been checking to see when it is added to the CA so I can confirm its working as it should.

This was the first time I have attempted a template for unraid, but it was simple and quick, so I imagine it will work just fine once the correct XML code is in place.

**Comment by Cobbert** [comment]
Just keep an eye on the recently added section and you should see it pop back up, I was there 5 hours ago but with the wrong xml that uses an incorrect docker-hub image.

**Comment by derneuere** [comment]
I pushed the updated XML yesterday and the doc says, that it should update within a couple of hours. Would be great if you could test it again 👍

**Comment by Cobbert** [comment]
Just checked the CA again, no sign of the template, I will keep the page open and check back frequently and will test again once its live again.

**Comment by jeremytodd1** [comment]
I'm still not seeing it in Community Apps either. I checked a few times yesterday as well and nothing.

**Comment by Cobbert** [comment]
Just rechecked the CA, still not seeing the template. Will keep checking, for now anyone looking to run with docker on unraid you could possible use the docker-compse add-on for unraid and write in a compose yml for your unraid environment.

**Comment by derneuere** [comment]
It was an invalid XML and I got some help from a moderator of the forum: https://github.com/LibrePhotos/librephotos-unraid/pull/1

I think it should be online in an hour or two.

**Comment by Cobbert** [comment]
Wonderful! I will have a look once I get home from work and if its there, I will run a test. Glad it was an easy fix and you got someone who knows more then we do!

**Comment by Cobbert** [comment]
It is live, I am pulling it now, all settings looked perfect when adding it into unraid, so far so good.

**Comment by jeremytodd1** [comment]
I just pulled the image as well. Haven't actually gone into it quite yet.

One thing I want to point out is that it didn't look like it setup the WebUI link correctly. I am not seeing a "WebUI" option in the container's drop down menu.

Image: **https://i.imgur.com/KqvLmEb.png**

**Comment by Cobbert** [comment]
Yes I noticed that too, Ill push a new XML with the link added, Think I just missed that part.

The system port is 3000 and you should be able to hit it at your unraid box ip port 3000, unless you changed the port of course.

**Comment by jeremytodd1** [comment]
Yep, going to :3000 does bring up the WebUi like normal, but the XML should definitely have that as well.

I'm actually stuck on the login page though. I'm assuming the default login is admin:admin, right? That is what I'm attempting, at least.

I check the logs and this is the error that is shown when clicking the "Login" button:

2022-08-10 16:00:55.066 CDT [952] ERROR:  relation "api_user" does not exist at character 714

**Comment by Cobbert** [comment]
OK, I submitted a few pull requests to add the proper URL schema to the XML.
[WebUI link](https://github.com/LibrePhotos/librephotos-unraid/pull/2)
[Dark Icon](https://github.com/LibrePhotos/librephotos-unraid/pull/4)

New to github, but I think I went about that correctly.

**Comment by Cobbert** [comment]
Let me start mine up and see what happens.

**Comment by jeremytodd1** [comment]
Actually I got that issue solved. It was a user error, not container error.

A long while back I did have a Librephotos container setup. I had a left over folder from that instance. After removing that and starting this new container up again I was able to register a new user.

It's now importing my photos successfully it looks like. Great!

**Comment by Cobbert** [comment]
My install is also working just fine, after having to delete the old folder from first install. Account was created with no issues and photos scanned in.

**Comment by Cobbert** [comment]
To get the link to work in the drop down menu, edit the container and add this to WebUI `http://[IP]:[PORT:80]`

**Comment by jeremytodd1** [comment]
Yep! I know how to fix that but the template should have that included, is all I'm saying.

**Comment by derneuere** [comment]
Merged your PRs. If everything is alright, then I would l like to call this done :)

**Comment by Cobbert** [comment]
Ill pull it one more time just to verify its all up to date, but it looks good at first glance.

**Comment by Cobbert** [comment]
OK, found one more error, In CA we had a warning about WebUI port mapping, Unraid wants the container port mapped to the WebUI in the template in order to automatically map to whatever host port the end user chooses when adding to unriad from the template.

Push [Here](https://github.com/LibrePhotos/librephotos-unraid/pull/5)

More info [Here](https://forums.unraid.net/topic/91621-docker-template-errors/)

after that is fixed, everything should be up to the standards that unraid is asking for.

Notification from unraid in CA
`Port referenced in webUI does not correspond with a container port defined. Port Referenced does not exist in Config`

**Comment by Plimmer** [comment]
Clean install, nothing in my appdata folder and I get this from my log:

```
PostgreSQL init process complete; ready for start up.

2022-08-11 20:27:31.247 CEST [256] LOG: starting PostgreSQL 13.7 (Ubuntu 13.7-0ubuntu0.21.10.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.2.0-7ubuntu2) 11.2.0, 64-bit
2022-08-11 20:27:31.247 CEST [256] LOG: listening on IPv4 address "127.0.0.1", port 5432
2022-08-11 20:27:31.247 CEST [256] LOG: could not bind IPv6 address "::1": Cannot assign requested address
2022-08-11 20:27:31.247 CEST [256] HINT: Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2022-08-11 20:27:31.252 CEST [256] LOG: listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2022-08-11 20:27:31.263 CEST [548] LOG: database system was shut down at 2022-08-11 20:27:31 CEST
2022-08-11 20:27:31.272 CEST [256] LOG: database system is ready to accept connections
2022-08-11 20:27:31.894 CEST [572] FATAL: role "librephotos" does not exist
2022-08-11 20:27:32.656 CEST [652] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:27:32.656 CEST [652] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:27:32.656 CEST [652] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:27:32.656 CEST [652] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:27:34.293 CEST [810] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:27:34.293 CEST [810] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" IN ('ALLOW_REGISTRATION', 'ALLOW_UPLOAD', 'SKIP_PATTERNS', 'HEAVYWEIGHT_PROCESS', 'MAP_API_KEY', 'IMAGE_DIRS')
2022-08-11 20:27:34.295 CEST [810] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:27:34.295 CEST [810] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:27:34.296 CEST [810] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:27:34.296 CEST [810] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:27:34.346 CEST [813] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:27:34.346 CEST [813] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 20:27:35.166 CEST [940] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:27:35.166 CEST [940] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 20:27:35.195 CEST [943] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:27:35.195 CEST [943] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 20:27:35.166 CEST [940] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:27:35.166 CEST [940] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 20:27:35.195 CEST [943] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:27:35.195 CEST [943] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 20:29:23.677 CEST [946] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:29:23.677 CEST [946] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:29:23.678 CEST [946] ERROR: relation "constance_config" does not exist at character 91

2022-08-11 20:29:23.678 CEST [946] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 20:29:23.692 CEST [946] ERROR: relation "api_user" does not exist at character 714

2022-08-11 20:29:23.692 CEST [946] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user" WHERE "api_user"."username" = 'admin' LIMIT 21
```

**Comment by Cobbert** [comment]
I got those errors too, but after letting it run for a minute, it seems to come online, think the errors are going to be another issue.

Mine is up and running, photos scan in too, here is my log from unraid.

```
PostgreSQL init process complete; ready for start up.

2022-08-11 12:12:57.423 PDT [80] LOG: starting PostgreSQL 13.7 (Ubuntu 13.7-0ubuntu0.21.10.1) on x86_64-pc-linux-gnu, compiled by gcc (Ubuntu 11.2.0-7ubuntu2) 11.2.0, 64-bit
2022-08-11 12:12:57.423 PDT [80] LOG: listening on IPv4 address "127.0.0.1", port 5432
2022-08-11 12:12:57.423 PDT [80] LOG: could not bind IPv6 address "::1": Cannot assign requested address
2022-08-11 12:12:57.423 PDT [80] HINT: Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2022-08-11 12:12:57.429 PDT [80] LOG: listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2022-08-11 12:12:57.440 PDT [783] LOG: database system was shut down at 2022-08-11 12:12:57 PDT
2022-08-11 12:12:57.452 PDT [80] LOG: database system is ready to accept connections
2022-08-11 12:12:57.913 PDT [791] FATAL: role "librephotos" does not exist
2022-08-11 12:13:02.076 PDT [863] ERROR: relation "constance_config" does not exist at character 91
2022-08-11 12:13:02.076 PDT [863] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" IN ('ALLOW_REGISTRATION', 'ALLOW_UPLOAD', 'SKIP_PATTERNS', 'HEAVYWEIGHT_PROCESS', 'MAP_API_KEY', 'IMAGE_DIRS')
2022-08-11 12:13:02.079 PDT [863] ERROR: relation "constance_config" does not exist at character 91
2022-08-11 12:13:02.079 PDT [863] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 12:13:02.080 PDT [863] ERROR: relation "constance_config" does not exist at character 91
2022-08-11 12:13:02.080 PDT [863] STATEMENT: SELECT "constance_config"."id", "constance_config"."key", "constance_config"."value" FROM "constance_config" WHERE "constance_config"."key" = 'ALLOW_UPLOAD' LIMIT 21
2022-08-11 12:13:02.220 PDT [866] ERROR: relation "api_user" does not exist at character 714
2022-08-11 12:13:02.220 PDT [866] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 12:13:02.268 PDT [868] ERROR: relation "api_user" does not exist at character 714
2022-08-11 12:13:02.268 PDT [868] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
2022-08-11 12:13:02.328 PDT [870] ERROR: relation "api_user" does not exist at character 714
2022-08-11 12:13:02.328 PDT [870] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user"
```

**Comment by Plimmer** [comment]
I have had it running for over an hour now, still get this error every time I try and log in with admin:admin

```
2022-08-11 21:46:10.287 CEST [1107] ERROR: relation "api_user" does not exist at character 714

2022-08-11 21:46:10.287 CEST [1107] STATEMENT: SELECT "api_user"."id", "api_user"."password", "api_user"."last_login", "api_user"."is_superuser", "api_user"."username", "api_user"."first_name", "api_user"."last_name", "api_user"."email", "api_user"."is_staff", "api_user"."is_active", "api_user"."date_joined", "api_user"."scan_directory", "api_user"."confidence", "api_user"."image_scale", "api_user"."semantic_search_topk", "api_user"."avatar", "api_user"."transcode_videos", "api_user"."nextcloud_server_address", "api_user"."nextcloud_username", "api_user"."nextcloud_app_password", "api_user"."nextcloud_scan_directory", "api_user"."favorite_min_rating", "api_user"."save_metadata_to_disk", "api_user"."datetime_rules", "api_user"."default_timezone" FROM "api_user" WHERE "api_user"."username" = 'admin' LIMIT 21
```

**Comment by Cobbert** [comment]
Are you sure you have nothing in your appdata folder? This sounds like when I tried to reinstall the docker container with existing librephotos folder in my appdata folder.

You should be getting a form to create a new user on the first launch.

On a fresh install, the screenshot below is what you should be seeing.
[Screenshot](https://imgur.com/a/ZeTGmsD)

If you do not see it, and it is indeed a fresh install then try hitting refresh a few times, first time around it seemed like my browser was holding the login page in the cache.

**Comment by Cobbert** [comment]
I think we are good to call this closed as everything seems to be working on the template side of things.

**Comment by Plimmer** [comment]
A restart of the docker was what I needed. I would like to be able to change the amount of workers in the settings maybe?

**Comment by Cobbert** [comment]
That would most likely require something to be passed via extra parameters in the template, I can easily add it in if someone can provide what needs to be passed to docker. I do not use librephotos on my unraid box, and I have made that change only via docker .env file.

**Comment by Cobbert** [comment]
This seems to be the script that is ran by the singleton docker.

[Script](https://github.com/LibrePhotos/librephotos-linux/blob/main/install-librephotos.sh)

You can get an idea of what the container is doing when you add it [here](https://github.com/LibrePhotos/librephotos-linux/blob/main/Dockerfile)

I don't see where the workers would be configured, someone else will need to help to solve that, and I am not sure if its as simple as just passing the config to docker via the template as of right now.

**Comment by derneuere** [comment]
> A restart of the docker was what I needed. I would like to be able to change the amount of workers in the settings maybe?

You can change the number of heavyweight workers in the admin settings. These are the workers who process your images. The number of API workers is currently fixed on the singleton version. Feel free to open up an issue in the librephotos-linux repo about it :)

**Comment by derneuere** [comment]
Very cool to have this feature now 👍 Thanks again, @Cobbert
